### PR TITLE
feat: Update Airflow to 2.5.3

### PR DESCRIPTION
**Pull request by mikaeld**
# Description
Update Airflow to 2.5.3

Notable changes :point_down: 
**New in 2.4** :sparkles: ([Announcement](https://airflow.apache.org/blog/airflow-2.4.0/) and [release notes](https://airflow.apache.org/docs/apache-airflow/2.4.0/release_notes.html#id174))
- [Data-aware Scheduling](https://airflow.apache.org/docs/apache-airflow/2.4.0/concepts/datasets.html#data-aware-scheduling): new Dataset class that can possibly replace our use of `ExternalTaskSensor` . This also comes with a new Dataset view in the UI to show dataset dependencies
- Auto-registered DAGs in a context manager: `with DAG(...) as dag:` becomes `with DAG(...)`:
- Auto-refresh on the home page
- Support for `TaskGroup` in `ExternalTaskSensor`

**New in 2.5** :sparkles: ([Announcement](https://airflow.apache.org/blog/airflow-2.5.0/) and [release notes](https://airflow.apache.org/docs/apache-airflow/2.5.0/release_notes.html#id191))

- Add annotations/notes to task instances and DAG runs
- "Parsed at" date added to DAG code view to tell users when the DAG was last updated
- QoL improvements to the Datasets UI view: Filter by date and global search by name
- Taskflow API support for Datasets (`@task` decorator)
- New trigger rule `one_done`

# Validation
https://dev.telemetry-airflow.nonprod.dataservices.mozgcp.net/home
image

# Related PR
https://github.com/mozilla-it/dataservices-infra/pull/49

### CODE REVIEW

File: dags/copy_deduplicate.py [hunk 1]
```diff
@@ -89,8 +88,8 @@
             "telemetry_live.first_shutdown_v4",
             "telemetry_live.saved_session_v4",
         ],
-        node_selectors={"nodepool": "highmem"},
-        resources=resources,
+        node_selector={"nodepool": "highmem"},
```
Comment (haroldwoo) [inline dags/copy_deduplicate.py:91]: they really changed the param names for GkePodOperator or the k8sPodOperator?

File: dags/glam_subdags/extract.py [hunk 2]
```diff
@@ -21,13 +23,13 @@ def extracts_subdag(
         SubDagOperator(
             subdag=extract_channel_subdag(
                 dag_id,
-                "extract_{}".format(channel),
+                f"extract_{channel}",
```
Comment (haroldwoo) [inline dags/glam_subdags/extract.py:26]: just curious, is your linter doing the f string swaps?

File: dags/operators/backport/__init__.py
Comment (haroldwoo) [inline dags/operators/backport/__init__.py:1]: should we rm the backport folder entirely?

**Review by haroldwoo — APPROVED** [review]

File: dags/glam_subdags/extract.py (hunk 2 above)
Comment (mikaeld) [reply dags/glam_subdags/extract.py:26]: yes

Comment (mikaeld) [reply dags/glam_subdags/extract.py:26]: E.g. it was also complaining about nested `if`s here, which is why I refactored it
image

**Pull request merged**